### PR TITLE
Add esf file parser to generate the node directories automatically

### PR DESCRIPTION
**Pull request by hhou2011**
@Pacane @d-shapiro

### CODE REVIEW

File: src/main/java/org/dsa/iot/knx/DeviceFolder.java [hunk 1]
```diff
@@ -78,7 +87,110 @@ protected void addPoint(ActionResult event) {
...
+	}
+
+	private PointType getDataTypeByDataSize(String measurement) {
+		PointType type;
+		switch (measurement) {
+		case "1 Bit":
+			type = PointType.BOOL;
+			break;
+		case "4 Bit":
+			type = PointType.CONTROL;
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/DeviceFolder.java:181]: missing breaks in there

File: src/main/java/org/dsa/iot/knx/KnxIpConnection.java [hunk 2]
```diff
@@ -170,61 +185,47 @@ public void handle(ActionResult event) {
...
-		public void addListener(KnxConnection listener) {
-			this.listener = listener;
 		}
 
-		@Override
-		public void run() {
-			try {
-				knxLink = new KNXNetworkLinkIP(serviceMode, localEP, remoteEP, useNat, TPSettings.TP1);
-				knxLink.addLinkListener(new NetworkLinkListener() {
+		public void linkClosed(CloseEvent e) {
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/KnxIpConnection.java:194]: Java's convention for methods like these would be `onLinkClosed`, `onConfirmation`, etc..

File: src/main/java/org/dsa/iot/knx/KnxIpConnection.java [hunk 3]
```diff
@@ -170,61 +185,47 @@ public void handle(ActionResult event) {
...
-					}
+	private class ProcessCommunicatorListener implements ProcessListener {
 
-				});
-			} catch (KNXException e1) {
-				LOGGER.debug("error:", e1);;
-			} catch (InterruptedException e1) {
-				LOGGER.debug("error:", e1);;
-			}
+		public void groupWrite(ProcessEvent e) {
```
Comment (Pacane) [inline src/main/java/org/dsa/iot/knx/KnxIpConnection.java:206]: same here

File: src/main/java/org/dsa/iot/knx/KnxIpConnection.java (hunk 2 above)
Comment (hhou2011) [inline src/main/java/org/dsa/iot/knx/KnxIpConnection.java:194]: These interface namings come from the third party library used in the project.

**Pull request merged**